mavis/ExtensionManager.hpp

```diff
@@ -114,6 +132,43 @@ namespace mavis::extension_manager
             }
     };
 
+    class ExtensionNotAllowedException : public ExtensionManagerException
+    {
+        private:
+            static inline std::string setToString_(const std::unordered_set<std::string>& set)
+            {
+                std::string result("[");
+                bool first = true;
+                for(const auto& str: set)
+                {
+                    if(first)
+                    {
+                        first = false;
+                    }
+                    else
+                    {
+                        result += ',';
+                    }
+
+                    result += str;
+                }
```

Comment:
Another way to do this:
```suggestion
                std::string comma = "";
                for(const auto& str: set)
                {
                    result += comma;
                    result += str;
                    comma = ',';
                }
```